### ISSUE TITLE
Using enthistory together with entgql causes problems

### ISSUE THREAD

**Issue by nixxxon** [flume/enthistory#37]
**Describe the bug**
I'm having some issues using enthistory together with [entgql](https://pkg.go.dev/entgo.io/contrib/entgql). Please close this issue if you think it's more related to entgql or bad configuration.

First problem:

```
failed to run ent codegen: execute template "client": template: node.tmpl:315:16: executing "client/fields/additional" at <gqlIDType (filterNodes $.Nodes (skipMode "type")) $.IDType>: error calling gqlIDType: entgql: mixed id types must be type string or implement the graphql.Marshaller/graphql.Unmarshaller interfaces
```
Seems to be related to the id field being skipped in the history entity since error goes away when commenting out the id check:
```go
for _, field := range original.Fields() {
	// if field.Descriptor().Name != "id" {
		historyFields = append(historyFields, field)
	// }
}
```

Second problem:
```
failed to load schema: internal/graphql/schema/query.graphql:6: Undefined type UUID.
```

Adding the type annotation to the ref field makes the error go away. I suspect there should be some way to configure entgql to use ID for all UUID fields, probably not a problem in enthistory..

```go
field.UUID("ref", uuid.UUID{}).
	Immutable().
	Optional().
	Annotations(
		entgql.Annotation{
			Type: "ID",
		},
	),
```

Third problem:

```
cannot use (*enthistory.OpType)(nil) (value of type *enthistory.OpType) as graphql.Unmarshaler value in variable declaration: *enthistory.OpType does not implement graphql.Unmarshaler (missing method UnmarshalGQL)
```

Adding the methods to OpType makes the error go away:
```go
func (op OpType) MarshalGQL(w io.Writer) {
	io.WriteString(w, op.String())
}

// UnmarshalGQL implements graphql.Unmarshaler interface.
func (op *OpType) UnmarshalGQL(val interface{}) error {
	str, ok := val.(string)
	if !ok {
		return fmt.Errorf("enum %T must be a string", val)
	}
	*op = OpType(str)

	switch *op {
	case OpTypeInsert, OpTypeUpdate, OpTypeDelete:
		return nil
	default:
		return fmt.Errorf("%s is not a valid history operation type", str)
	}
}
```

**To Reproduce**
https://github.com/nixxxon/entdemo 

Start from the first commit and run `go generate .` to get the first problem.

**Expected behavior**
The history entities and GraphQL schema being successfully generated.

**Additional context**
Thanks for all the great work with this.

**Comment by frisbm** [comment]
@nixxxon Thanks for opening an issue. I'll take a look at these today and try to get a fix in, the bug report repo should help speed this along

**Comment by frisbm** [comment]
@nixxxon I think i got those issues resolved in #38 if you want to take a look. 
* Mixed Id Types issue required creating a new config for inheriting Id Types on the history models from their original schemas, effectively what you presented
* Bad typing in the graphql schema for uuid was actually an issue with enthistory, just needed to get the ID annotations in the right spot like you showed
* Optype unmarshalling was a simple as using your snippet

Feel free to pull this branch into your project and give it a try

**Comment by nixxxon** [comment]
> @nixxxon I think i got those issues resolved in #38 if you want to take a look.
> 
> * Mixed Id Types issue required creating a new config for inheriting Id Types on the history models from their original schemas, effectively what you presented
> * Bad typing in the graphql schema for uuid was actually an issue with enthistory, just needed to get the ID annotations in the right spot like you showed
> * Optype unmarshalling was a simple as using your snippet
> 
> Feel free to pull this branch into your project and give it a try

I can confirm that using your branch fixes my problems, good job and thank you!
